tests/kafkatest/tests/consumer_test.py

```diff
@@ -292,28 +231,75 @@ def test_broker_failure(self, clean_shutdown, enable_autocommit):
             assert consumer.last_commit(partition) == consumer.current_position(partition), \
                 "Last committed offset did not match last consumed position"
 
-    def test_simple_consume(self):
-        total_records = 1000
+    def test_group_consumption(self):
+        """
+        Verifies correct group rebalance behavior as consumers are started and stopped. 
+        In particular, this test verifies that the partition is readable after every
+        expected rebalance.
 
-        consumer = self._setup_consumer(self.STOPIC)
-        producer = self._setup_producer(self.STOPIC, max_messages=total_records)
+        Setup: single Kafka cluster with a group of consumers reading from one topic
+        with one partition while the verifiable producer writes to it.
 
-        partition = TopicPartition(self.STOPIC, 0)
+        - Start the consumers one by one, verifying consumption after each rebalance
+        - Shutdown the consumers one by one, verifying consumption after each rebalance
+        """
+        consumer = self.setup_consumer(self.TOPIC)
+        producer = self.setup_producer(self.TOPIC)
 
-        consumer.start()
-        self._await_all_members(consumer)
+        partition = TopicPartition(self.TOPIC, 0)
 
         producer.start()
-        wait_until(lambda: producer.num_acked == total_records, timeout_sec=20,
-                   err_msg="Producer failed waiting for messages to be written")
 
-        wait_until(lambda: consumer.last_commit(partition) == total_records, timeout_sec=10,
-                   err_msg="Consumer failed to read all expected messages")
+        num_started = 0
```

Comment:
Only annoying thing about `enumerate` in this case is that it starts from 0. So it looks like this in code:
```python
for i, node in enumerate(consumer.nodes):
    consumer.start_node(node)
    self.await_members(consumer, i+1)
    self.await_consumed_messages(consumer)
```
Guess it's more concise, if a little less clear.